### ISSUE TITLE
Crash with a player(internal) recipe book render

### ISSUE THREAD

**Issue by AssB0ys** [Vazkii/Patchouli#126]
Hello.
I have a problem with the mod patchuoli. It took me a lot of time to point out which mod causes this and finally after disabling mods one by one and relaunching the game I was able to detrermine the following behavious with patchouli mod:
- First strange behavious is when you:
  1. Open an player inventory.
  2. Click book icon so that recipes tab appear.
  3. close inventory.
  4. Open player inventory again and the recipe tab is not active.

This does not happen with the crafting table inventory. Now that may sound like nothing much and I agree. I wrote it because it may be related to my main issue that I will explain in the next paragraph.

Sometimes when I try to craft things in a player inventroy(2x2 crafting grid) the game crashes with a following message https://pastebin.com/pSt4ZPL6
Recipes from some mobs have nearly a 100% chance of crash, for example Vanilla Food Pantry, that's why I even thought that VFP was causing this crash at first. But then I took my time and was disabling mods one by one and relaunching the game(more than 90 game launches, it took me three evenings :-D). The crash happens with even vanilla recipes but not 100% of time, as I said I sometimes am able to craft a crafting table, sometimes even the crafting table recipe is crashing the game.
1. Open player inventory
2. Put things in a 2x2 crafting grid
3. Take an item out of a result slot so that it is actually on your mouse cursor
4. On this step the crash happens. I am able to move the cursor a couple millimeters and then the game starts to shut down its internal servers.

Tested it with the latest forge (14.23.5.2835) and forge version 14.23.5.2795 because this version is recommended by optifine (at some point I thought the optifine was causing this, so I had to test this too). And any older than the current latest that may have been the latest at days I was testing this(I've been troubleshooting for past two weeks). Thank you. The game version is 1.12.2.
Also in "patchouli.cfg" file I've noticed the following:
```
    # Set this to the ID of a book to have it show up in players' inventories, replacing the recipe book.
    S:"Inventory Button Book"=
```
full log at https://pastebin.com/eLMFbJzp

**Comment by AssB0ys** [comment]
Additional information:
I was told(recommended) to use FastWorkbench mod to avoid the issue and the workaround seems to work. The game does not crash. And if you enable the recipe book back in FWB config the crash happens. This is just my observation, maybe it helps.

**Comment by AssB0ys** [comment]
This is also happening without optifine and phosphor installed.
Full error log without optifine and installed:
https://pastebin.com/UxwNAtf6

**Comment by Amagous** [comment]
Has there been any update on fixing this? I am having the same problem.

Pastebin to crashlog: https://pastebin.com/Cg2PnT4z

**Comment by ticviking** [comment]
I've been tracking this down myself in. #154

could you try a new instance with just patchouli and see if you can reproduce this also.

**Comment by williewillus** [comment]
fixed in latest 1.12 build
